tests/test_lib_usercache.py

```diff
@@ -453,6 +454,23 @@ def test_12_multiple_resolvers(self):
         delete_resolver('reso_a')
         delete_resolver('reso_b')
 
+    def test_13_cache_username(self):
+
+        self.counter = 0
+
+        def get_username(uid, resolver):
+            self.counter += 1
+            return "user1"
+
+        r = cache_username(get_username, "uid1", "reso1")
+        self.assertEqual(r, "user1")
+        self.assertEqual(self.counter, 1)
+
+        # The second call does not increase the counter, since the result is fetched from the cache
+        r = cache_username(get_username, "uid1", "reso1")
+        self.assertEqual(r, "user1")
+        self.assertEqual(self.counter, 1)
```

Comment:
Hm, I think the unit tests only cover the case that ``user.used_login == user.login``. Could you add another test for the case that they differ?
... but I guess that's not so easy because the unittest uses the SQL resolver which doesn't support multiple login attributes :-/

Reply:
True. I would rather like to refrain from doing this.